### PR TITLE
Add test hook runner for plugins

### PR DESCRIPTION
**Pull request by alanking**
In progress, but this works for plugins which don't take extra arguments (e.g. audit)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,6 +84,35 @@ Running the federation test suite is very similar. Note: The federation test sui
...
+            - centos:7      ->  `Centos linux_7`
+    - iRODS server is already installed and setup
+
+Your provided built packages should be in an identical directory or symlink following the naming convention above.
+
+In the future, these requirements will be relaxed so that creating a build-and-test workflow will not be as difficult.
+
+### How to run a test hook
+
+We will use the curl microservice plugin as an example. The curl microservice plugin package is assumed to have been built for the target platform. The last argument provided to the script is the name of the git repository from which the test hook will be fetched.
```
Comment (trel) [inline README.md:107]: "git repository"   ... but it's not in the example below... it's a local directory?

File: test_utils.py [hunk 2]
```diff
@@ -80,7 +91,12 @@ def run_specific_tests(container, tests, options=None, fail_fast=True):
 
 
 def run_python_test_suite(container, options=None):
-    # start constructing the run_tests command
+    """Run the entire python test suite for iRODS.
+
+    Arguments:
+    container -- target container on which the test script will run
+    options -- list of strings representing script options to pass to the run_tests.py script
```
Comment (trel) [inline test_utils.py:98]: run_core_tests.py

File: test_utils.py [hunk 3]
```diff
@@ -96,3 +112,33 @@ def run_python_test_suite(container, options=None):
...
     return ec
+
+
+def run_test_hook(container, repo_name, branch=None, options=None):
+    """Run the test hook from the specified git repository.
+
+    Arguments:
+    container -- target container on which the test script will run
+    repo_name -- name of the git repo
+    options -- list of strings representing script options to pass to the run_tests.py script
```
Comment (trel) [inline test_utils.py:123]: same

File: README.md (hunk 1 above)
Comment (alanking) [reply README.md:107]: The last argument is `irods_microservice_plugins_curl` which is the name of the git repository. The local directory - `/path/to/curl-microservice-plugin` - refers to the location of the built packages.

File: test_utils.py (hunk 2 above)
Comment (alanking) [reply test_utils.py:98]: This is referring to `run_tests.py` script which exists in the irods repository under the scripts directory

File: projects/centos-7/Dockerfile [hunk 4]
```diff
@@ -19,4 +19,11 @@ RUN rpm --import https://packages.irods.org/irods-signing-key.asc && \
     wget -qO - https://packages.irods.org/renci-irods.yum.repo | tee /etc/yum.repos.d/renci-irods.yum.repo && \
     yum update -y
 
+# TODO: For some reason, pip does not work from installation above...
```
Comment (trel) [inline projects/centos-7/Dockerfile:22]: seems weird.

Comment (alanking) [reply projects/centos-7/Dockerfile:22]: Indeed. I made #25 for this

File: README.md (hunk 1 above)
Comment (alanking) [reply README.md:107]: We can hash out some of these options in the future. WIP :)

File: test_utils.py (hunk 2 above)
Comment (alanking) [reply test_utils.py:98]: This is as it should be.

**Pull request merged**